test/Interop/Cxx/ergonomics/implicit-computed-properties-module-interface.swift

```diff
@@ -0,0 +1,258 @@
+
+// CHECK:      struct IntGetterSetter {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var x: Int32
+// CHECK-NEXT:    func getX() -> Int32
+// CHECK-NEXT:    mutating func setXMutating(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct IntGetterSetterSnakeCaseUpper {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    func Get_X() -> Int32
+// CHECK-NEXT:    mutating func Set_X(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct IntGetterSetterSnakeCase {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    func get_x() -> Int32
+// CHECK-NEXT:    mutating func set_x(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct GetterHasArg {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    func getX(_ v: Int32) -> Int32
+// CHECK-NEXT:    mutating func setXMutating(_ v: Int32)
+// CHECK-NEXT: }
+
+// CHECK:      struct GetterSetterIsUpper {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var x: Int32
+// CHECK-NEXT:    func GETX() -> Int32
+// CHECK-NEXT:    mutating func SETX(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct HasXAndY {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var xAndY: Int32
+// CHECK-NEXT:    func GetXAndY() -> Int32
+// CHECK-NEXT:    mutating func SetXAndY(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct AllUpper {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var fooandbar: Int32
+// CHECK-NEXT:    func GETFOOANDBAR() -> Int32
+// CHECK-NEXT:    mutating func SETFOOANDBAR(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct BothUpper {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var fooAndBAR: Int32
+// CHECK-NEXT:    func getFOOAndBAR() -> Int32
+// CHECK-NEXT:    mutating func setFOOAndBAR(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct FirstUpper {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var fooAndBar: Int32
+// CHECK-NEXT:    func getFOOAndBar() -> Int32
+// CHECK-NEXT:    mutating func setFOOAndBar(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct NonConstGetter {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var x: Int32 { mutating get set }
+// CHECK-NEXT:    mutating func getXMutating() -> Int32
+// CHECK-NEXT:    mutating func setXMutating(_ v: Int32)
+// CHECK-NEXT:    var val: Int32
+// CHECK-NEXT: }
+
+// CHECK:      struct ConstSetter {
+// CHECK-NEXT:    init()
+// CHECK-NEXT:    init(val: Int32)
+// CHECK-NEXT:    var x: Int32 { mutating get set }
+// CHECK-NEXT:    mutating func getX() -> Int32
```

Comment:
Interesting: so these both got imported as mutating. We need a test where these are not imported as mutating.

Reply:
You can either create a global, or (probably better) just write a class like this with an empty setter:
```
struct ConstGetterAndSetter {
   int getfoo() const { return 42; }
   void setfoo(int v) const { }
 };
```
Still test the setter, but know the value won't change. That's OK.